crates/sui-core/src/transaction_driver/mod.rs

```diff
@@ -209,11 +324,12 @@ where
     #[instrument(level = "error", skip_all, err)]
     async fn drive_transaction_once(
         &self,
-        tx_digest: &TransactionDigest,
+        tx_digest: Option<TransactionDigest>,
```

Comment:
Optional: I think there is always one Some() and one None in `tx_digest` and `ping`. Maybe we can drop the `ping` arg and just reply on `tx_digest: Option<TransactionDigest>` and `tx_type: TxType` to decide the `PingType`.